force-app/main/default/classes/CON_ContactMerge_CTRL.cls

```diff
@@ -950,98 +950,137 @@ public with sharing class CON_ContactMerge_CTRL {
     * @return PageReference The page that it redirects to. Same page user is in.
     */
     public PageReference mergeContacts() {
-        SObject master;
-        // Find the master record based the selected value of the Master FieldRow
+        Contact winningContact = getWinningContact();
         
+        if (winningContact == null) {
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error, label.npe01.Contact_Merge_Error_No_Master));
+            return null;
+        }
+
+        populateContactWithNonUniqueFields(winningContact);
+
+        List<Contact> losingContacts = groupLosingContacts(winningContact);
+        
+        // before proceeding further first lock the records for change
+        List<Contact> allContacts = new List<Contact>();
+        allContacts.add((Contact)winningContact);
+        allContacts.addAll(losingContacts);
+
+        List<Contact> lockedContacts = lockAllContacts(allContacts);
+
+        if (lockedContacts == null || lockedContacts.size() != allContacts.size()) {
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error, 
+                label.npe01.Contact_Merge_Error_Lock_failed  + ' ' + 
+                label.npe01.Contact_Merge_Error_please_retry) );
+            return null;
+        }
+
+        System.SavePoint sp = Database.setSavepoint();
+        try {
+            // We need to turn off Affiliation processing or there will be duplicate affiliates
+            // created after the merge.  As merge transactions do not fire child triggers
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.AFFL_doNotRunForContacts, true);
+            updateAndHandleWinningContact(winningContact);
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.AFFL_doNotRunForContacts, false);
+            
+            CON_ContactMerge.mergeContacts(winningContact, losingContacts);
+
+            winningContact = populateContactWithOnlyUniqueFields(winningContact.Id);
+            updateAndHandleWinningContact(winningContact);
```

Comment:
@andrewyu-salesforce Would it be possible to only call this if a field was changed? Merging Unique fields will be pretty uncommon...

Reply:
The boolean check is added